docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+# `ootb-variants-example-dev` where `ootb-variants` is the project name and
+# `example-dev` is the variant name.
+project-name = "ootb-variants"
+
+# This allows the maintainer to rev a version globally for all variants defined in
+# the project. We may choose to enforce an opinionated versioning scheme here or
+# allow any arbitrary string, TBD.
+project-version = "1.0.0"
+
+# The container image with the SDK, buildsys, etc. This is provided as a structure
+# so that we can switch on the target architecture. For example, when building
+# Bottlerocket images for x86_64 machines, the following would be produced:
+# public.ecr.aws/bottlerocket/bottlerocket-build-tools-x86_64:v1.14.1. Twoliter
+# will assume the container is available locally if the registry field is null
+# or an empty string.
+[sdk-image]
+registry = "public.ecr.aws/bottlerocket"
+name = "bottlerocket-sdk"
+version = "v0.30.0"
+
+# This is the container that Twoliter uses to invoke `cargo make` and other build
+# commands. It will be available for both x86_64 and Arm hosts, but it is agnostic
+# as to which architecture the build is targeting.
+[twoliter-image]
+registry = "public.ecr.aws/bottlerocket"
+name = "twoliter"
+version = "v0.1.0"
+```
+
+### Directory Structure
+
+Project directory structure is similar to what we see in Bottlerocket's main repo:
+
+```
+.
+├── build
+├── kits
+│   └── hello-dev-kit
+│       ├── src
+│       │   ├── kit.rs
+│       │   └── kit.spec
+│       └── Cargo.toml
+├── packages
+│   └── hello-agent
+│       ├── build.rs
+│       ├── Cargo.toml
+│       ├── hello-agent.service
+│       ├── hello-agent.spec
+│       └── pkg.rs
+├── sources
+│   ├── hello-agent
+│   │   ├── src
+│   │   │   └── main.rs
+│   │   └── Cargo.toml
+│   ├── Cargo.toml
+│   ├── clarify.toml
+│   └── deny.toml
+├── variants
+│   ├── example-dev
+│   │   ├── build.rs
+│   │   ├── Cargo.toml
+│   │   ├── pkg.rs
+│   │   └── variant.spec
+│   └── Cargo.toml
+├── COPYRIGHT
+├── LICENSE-APACHE
+└── LICENSE-MIT
+```
+
+Directory descriptions:
+
+* build: The output directory for Bottlerocket images, RPM files, etc.
+* kits: Where the local kits are defined.
+* packages: Where the RPM packages are defined.
+* variants: Where Bottlerocket image builds are defined.
+
+### Twoliter New
+
+Inspired by `cargo new`, we will implement a `twoliter new` command that creates a new Twoliter project.
+The command creates a directory structure like the one shown above.
+The new project will include a minimal variant with a simple custom package and settings.
+The command will take a few arguments, such as `--project name` and `--variant-name`.
+
+Similarly, a command will be available to add a new variant to an existing project.
+For example, `twoliter new-variant` which will add a variant.
+
+```
+❯ twoliter new project --name ootb-variants --variant-name example-dev
+❯ cd ootb-variants
+❯ twoliter build --variant hello-dev
+  ...
+❯ twoliter new-variant my-new-variant
```

Comment:
Can we avoid options and use sub-commands / sub-sub-commands? E.g.
* `twoliter build variant ...`
* `twoliter new project --name ... --variant ...`
* `twoliter new variant ...`